### PR TITLE
Remove indents for code style of file-scope based

### LINKED ISSUE #172: Change all templates to file-scoped namespace style

### PR DESCRIPTION
**Pull request by Hanazawa**
Resolve #172 

Note that I'm not familiar with MVC and Blazor UI stuff. Please feel free to test and I will fix it asap if any changes that breaks your code lol

### CODE REVIEW

File: src/AbpHelper.Core/Templates/Service/Groups/Service/src/{{ProjectInfo.FullName}}.Application/{{Option.Folder}}/{{Option.Name}}AppService.cs [hunk 1]
```diff
@@ -1,10 +1,8 @@
 {{~ if Option.Folder ~}}
-namespace {{ ProjectInfo.FullName }}.{{ Option.Folder | string.replace "/" "." }}
+namespace {{ ProjectInfo.FullName }}.{{ Option.Folder | string.replace "/" "." }} ;
```
Comment (JadynWong) [inline src/AbpHelper.Core/Templates/Service/Groups/Service/src/{{ProjectInfo.FullName}}.Application/{{Option.Folder}}/{{Option.Name}}AppService.cs:2]: Should be remove the space before the semicolon.

File: src/AbpHelper.Core/Templates/Service/Groups/Service/src/{{ProjectInfo.FullName}}.Application.Contracts/{{Option.Folder}}/I{{Option.Name}}AppService.cs [hunk 2]
```diff
@@ -1,12 +1,11 @@
 using Volo.Abp.Application.Services;
 
 {{~ if Option.Folder ~}}
-namespace {{ ProjectInfo.FullName }}.{{ Option.Folder | string.replace "/" "." }}
+namespace {{ ProjectInfo.FullName }}.{{ Option.Folder | string.replace "/" "." }} ;
 {{~ else ~}}
-namespace {{ ProjectInfo.FullName }}
+namespace {{ ProjectInfo.FullName }} ;
```
Comment (JadynWong) [inline src/AbpHelper.Core/Templates/Service/Groups/Service/src/{{ProjectInfo.FullName}}.Application.Contracts/{{Option.Folder}}/I{{Option.Name}}AppService.cs:6]: Should be remove the space before the semicolon.

File: src/AbpHelper.Core/Templates/Crud/Groups/Repository/src/{{ProjectInfo.FullName}}.EntityFrameworkCore/{{EntityInfo.RelativeDirectory}}/{{EntityInfo.Name}}Repository.cs [hunk 3]
```diff
@@ -6,23 +6,30 @@
...
 using System;
+using System.Linq;
+using System.Threading.Tasks;
 using {{ ProjectInfo.FullName }}.EntityFrameworkCore;
 using Volo.Abp.Domain.Repositories.EntityFrameworkCore;
 using Volo.Abp.EntityFrameworkCore;
 
-namespace {{ EntityInfo.Namespace }}
-{
+namespace {{ EntityInfo.Namespace }} ;
```
Comment (JadynWong) [inline src/AbpHelper.Core/Templates/Crud/Groups/Repository/src/{{ProjectInfo.FullName}}.EntityFrameworkCore/{{EntityInfo.RelativeDirectory}}/{{EntityInfo.Name}}Repository.cs:15]: Should be remove the space before the semicolon.

File: src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs [hunk 4]
```diff
@@ -13,29 +13,29 @@
...
+{{~ end ~}}
+{{~ if defined_remote_service ~}}
+{{ defined_remote_service }}
+{{~ else ~}}
+[RemoteService(Name = "{{ ProjectInfo.Name }}{{ Option.Name }}")]
+{{~ end ~}}
+{{~ if ProjectInfo.TemplateType == 'Application' ~}}
+[Route("/api/app/{{ Option.Name | abp.kebab_case }}")]
+{{~ else if ProjectInfo.TemplateType == 'Module' ~}}
+[Route("/api/{{ ProjectInfo.Name | abp.kebab_case }}/{{ Option.Name | abp.kebab_case }}")]
```
Comment (JadynWong) [inline src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:27]: Undefined `kebab_case` method.

https://github.com/EasyAbp/AbpHelper.CLI/blob/ec35f00513d4b277b0dda1058e16ec6682734298/src/AbpHelper.Core/Generator/AbpFunctions.cs#L14-L23

image

File: src/AbpHelper.Core/Templates/Crud/Groups/Test/test/{{ProjectInfo.FullName}}.Domain.Tests/{{EntityInfo.RelativeDirectory}}/{{EntityInfo.Name}}DomainTests.cs [hunk 5]
```diff
@@ -2,24 +2,23 @@
...
+        // Assert
 
-            // Assert
-        }
-        */
+        // Assert
     }
+    */
+}
 }
```
Comment (JadynWong) [inline src/AbpHelper.Core/Templates/Crud/Groups/Test/test/{{ProjectInfo.FullName}}.Domain.Tests/{{EntityInfo.RelativeDirectory}}/{{EntityInfo.Name}}DomainTests.cs:24]: Should be remove `}`.

image

**Review by gdlcf88 — CHANGES_REQUESTED** [review]

File: src/AbpHelper.Core/Templates/Methods/Groups/Service/src/{{ProjectInfo.FullName}}.Application.Contracts/{{InterfaceInfo.RelativeDirectory}}/Dtos/{{Bag.Name}}Input.cs [hunk 6]
```diff
@@ -1,10 +1,8 @@
 {{- SKIP_GENERATE = Option.NoInput -}}
 using System;
 
-namespace {{ InterfaceInfo.Namespace }}.Dtos
+namespace {{ InterfaceInfo.Namespace }}.Dtos;
```
[the remark this answers is not in the document]
Comment (gdlcf88) [reply src/AbpHelper.Core/Templates/Methods/Groups/Service/src/{{ProjectInfo.FullName}}.Application.Contracts/{{InterfaceInfo.RelativeDirectory}}/Dtos/{{Bag.Name}}Input.cs:4]: > Question🤔: Should there be a blank line below `namespace` and `using`?
> 
> image image

image
I think so. I found many places that need to be modified.

File: src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs [hunk 7]
```diff
@@ -13,29 +13,29 @@
...
-        {{~ for method in ClassInfo.Methods | abp.intersect InterfaceInfo.Methods ~}}
-{{~ include "Templates/Controller/ControllerMethod" method ~}}
-        {{~ end ~}}
+    public {{ Option.Name }}Controller(I{{ Option.Name }}AppService service)
+    {
+        _service = service;
     }
+{{~ for method in ClassInfo.Methods | abp.intersect InterfaceInfo.Methods ~}}
+{{~ include "Templates/Controller/ControllerMethod" method ~}}
+    {{~ end ~}}
```
Comment (gdlcf88) [inline src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:39]:
```c#
    {{~ end ~}}
```
should be
```c#
{{~ end ~}}
```

File: src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs (hunk 4 above)
Comment (Hanazawa) [reply src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:27]: Thanks for testing.

I need your assist to decide which version of code I should do the copy & paste.

Previously I modified these codes base on [AbpHelper.GUI](https://github.com/EasyAbp/AbpHelper.CLI/blob/67bbde8d9f525820e63ca162a32263732ba16f05/src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/%7B%7BProjectInfo.FullName%7D%7D.HttpApi/%7B%7BInterfaceInfo.RelativeDirectory%7D%7D/%7B%7BOption.Name%7D%7DController.cs) at line 25, 27. And I found out the there are differences on [AbpHelper.CLI](https://github.com/EasyAbp/AbpHelper.CLI/blob/develop/src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/%7B%7BProjectInfo.FullName%7D%7D.HttpApi/%7B%7BInterfaceInfo.RelativeDirectory%7D%7D/%7B%7BOption.Name%7D%7DController.cs) at line 25, 27.

Comment (gdlcf88) [reply src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:27]: This is a problem. Maybe we should merge other PRs first.

Comment (gdlcf88) [reply src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:27]: @Hanazawa I am merging the PRs in https://github.com/EasyAbp/AbpHelper.CLI/labels/GUI-used-in-advance.

After I finish. We may need a new round of review to ensure that we have not modified the expired code in this PR.

Comment (gdlcf88) [reply src/AbpHelper.Core/Templates/Controller/Groups/Controller/src/{{ProjectInfo.FullName}}.HttpApi/{{InterfaceInfo.RelativeDirectory}}/{{Option.Name}}Controller.cs:27]: @Hanazawa I think we can merge the `EasyAbp:develop` branch into the current branch now.

**Review by JadynWong — APPROVED** [review]
LGTM.

- `{{DtoInfo.CreateTypeName}}.cs`
- `Create{{EntityInfo.Name}}ViewModel.cs`
- `CreateEdit{{EntityInfo.Name}}ViewModel.cs`
- `Edit{{EntityInfo.Name}}ViewModel.cs`

https://github.com/EasyAbp/AbpHelper.CLI/blob/ad4bcb8a05c07124b7983fabbcc9f1e6e2129bb7/src/AbpHelper.Core/Templates/Crud/Groups/Service/src/%7B%7BProjectInfo.FullName%7D%7D.Application.Contracts/%7B%7BEntityInfo.RelativeDirectory%7D%7D/Dtos/%7B%7BDtoInfo.CreateTypeName%7D%7D.cs#L1-L4

https://github.com/EasyAbp/AbpHelper.CLI/blob/ad4bcb8a05c07124b7983fabbcc9f1e6e2129bb7/src/AbpHelper.Core/Templates/Crud/Groups/UiRazor/src/%7B%7BProjectInfo.FullName%7D%7D.Web/Pages/%7B%7BBag.PagesFolder%7D%7D/%7B%7BEntityInfo.RelativeDirectory%7D%7D/%7B%7BEntityInfo.Name%7D%7D/ViewModels/CreateEdit%7B%7BEntityInfo.Name%7D%7DViewModel.cs#L1-L8

`{{~ if ... }}using System.ComponentModel;{{ end ~}}` position seems to cause the space to disappear or be duplicated.

This is not caused by this change.

image

image

**Comment by gdlcf88** [comment]
> This is not caused by this change.

We should check it later.

**Comment by gdlcf88** [comment]
Thanks @Hanazawa.

**Pull request merged**